src/plugins/data/common/index_patterns/index_patterns/flatten_hit.ts

```diff
@@ -78,7 +78,20 @@ function decorateFlattenedWrapper(hit: Record<string, any>, metaFields: Record<s
     // unwrap computed fields
     _.forOwn(hit.fields, function (val, key: any) {
       if (key[0] === '_' && !_.includes(metaFields, key)) return;
-      flattened[key] = Array.isArray(val) && val.length === 1 ? val[0] : val;
+      if (Array.isArray(val)) {
```

Comment:
I've made this verbose, but I'm open to more compact implementation, if someone has a suggestion.

Reply:
I'm not exactly sure what the purpose of this piece of code is... Maybe adding a clarifying comment will help. I understand that, if it's an empty array, it replaces it with undefined, or if it's a one-item array, it replaces it with the item itself rather than array, but I just don't understand *why*.

Reply:
Simplified and added a comment